### PR TITLE
Add external_id as indexable case property

### PR DESCRIPTION
**Pull request by wpride**
Enables indexed searches by an `external_id` property to be used by (you guessed it) external systems like SimPrints. The implementation uses the existing `external_id` case property as the value if it exists since this is already used by HQ in the same way. 

cross-request: https://github.com/dimagi/commcare-core/pull/814

### CODE REVIEW

**Comment by wpride** [comment]
@damagatchi retest this please

File: app/src/org/commcare/models/database/app/AppDatabaseUpgrader.java [hunk 1]
```diff
@@ -28,7 +28,6 @@
 
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.Comparator;
```
Comment (ctsims) [inline app/src/org/commcare/models/database/app/AppDatabaseUpgrader.java]: no-op

Comment (ctsims) [reply app/src/org/commcare/models/database/app/AppDatabaseUpgrader.java]: whoops, nevermind.

**Comment by ctsims** [comment]
This looks good to go once the version bump happens and you've had a chance to test the migration

**Comment by wpride** [comment]
Release Note: Make `external_id` property indexed

**Pull request merged**